docs/UserGuide.md

```diff
@@ -126,25 +134,43 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 Examples:
 * `find John` returns `john` and `John Doe`
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+  <!--- ![result for 'find alex david'](images/findAlexDavidResult.png) --->
+
+### Filtering residents by field : `filter`
+
+Shows a list of residents whose fields match the input keywords.
+
+Format: `filter [/specifier] KEY:VALUE [ADDITIONAL_KEYS:ADDITIONAL VALUES]`
+
+* The default specifier, if not specified by the user is `/all`.
+* Currently, only two specifiers are supported:
+  * `/all` returns a resident if it fulfills **all** of the specified keywords (Logical AND operation).
+  * `/any` returns a resident if it fulfills **any** of the specified keywords (Logical OR operation).
+* Repeated keys are not permitted for `/all` specifier, e.g. `filter /all house:Draco house:Aquila` will not work.
+* Valid keys are those included when [adding](#adding-a-resident--add) or [editing](#editing-an-existing-resident--edit) a resident e.g. `NAME, PHONE_NUMBER, EMAIL, FLOOR, UNIT, and TAG`
```

Comment:
Maybe instead of listing out all the fields, we can make a section about all fields that are stored by RC4HDB and add a link to the part there.